qiskit/synthesis/discrete_basis/solovay_kitaev.py

```diff
@@ -0,0 +1,257 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Synthesize a single qubit gate to a discrete basis set."""
+
+from __future__ import annotations
+
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, Gate
+from qiskit.converters import circuit_to_dag
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.dagcircuit.dagcircuit import DAGCircuit
+from qiskit.transpiler.exceptions import TranspilerError
+
+from .gate_sequence import GateSequence
+from .commutator_decompose import commutator_decompose
+from .generate_basis_approximations import generate_basic_approximations, _1q_gates, _1q_inverses
+
+
+class SolovayKitaevDecomposition:
+    """The Solovay Kitaev discrete decomposition algorithm.
+
+    This class is called recursively by the transpiler pass, which is why it is separeted.
+    See :mod:`qiskit.synthesis.discrete_basis` for more information.
+    """
+
+    def __init__(
+        self, basic_approximations: str | dict[str, np.ndarray] | list[GateSequence] | None = None
+    ) -> None:
+        """
+
+        .. note::
+
+            If ``basis_gates`` and ``depth`` are not passed, the basic approximations can be
+            generated with the ``generate_basic_approximations`` method and loaded into the
+            class via ``load_basic_approximations``. Since in practice, large basic approximations
+            are required we suggest to generate a sufficiently large set once and always load
+            the approximations afterwards.
+
+        Args:
+            basis_gates: The basis gates used in the basic approximations.
+            depth: The maximum depth of the basic approximations.
```

Comment:
`basic_approximations` isn't listed here. I would expect a docstring for this as it's a complex argument that defines a lot of the behavior. Also `depth` and `basis_gates` no longer exist.

Reply:
Oh, my bad. Weird that this was not catched by the linter though!